framework/db/SchemaBuilder.php

```diff
@@ -0,0 +1,412 @@
+    {
+        return static::createNumeric(Schema::TYPE_DOUBLE, $precision, $scale);
+    }
+
+    /**
+     * Makes column a decimal
+     *
+     * @param integer $precision
+     * @param integer $scale
+     * @return SchemaBuilder
+     */
+    public static function decimal($precision = null, $scale = null)
+    {
+        return static::createNumeric(Schema::TYPE_DECIMAL, $precision, $scale);
+    }
+
+    /**
+     * Makes column a datetime
+     *
+     * @param integer $length
+     * @return SchemaBuilder
+     */
+    public static function dateTime($length = null)
+    {
+        return static::createDefault(Schema::TYPE_DATETIME, $length);
+    }
+
+    /**
+     * Makes column a timestamp
+     *
+     * @param integer $length
+     * @return SchemaBuilder
+     */
+    public static function timestamp($length = null)
+    {
+        return static::createDefault(Schema::TYPE_TIMESTAMP, $length);
+    }
+
+    /**
+     * Makes column a time
+     *
+     * @param integer $length
+     * @return SchemaBuilder
+     */
+    public static function time($length = null)
+    {
+        return static::createDefault(Schema::TYPE_TIME, $length);
+    }
+
+    /**
+     * Makes column a date
+     *
+     * @return SchemaBuilder
+     */
+    public static function date()
+    {
+        return static::createDefault(Schema::TYPE_DATE);
+    }
+
+    /**
+     * Makes column a binary
+     *
+     * @param integer $length
+     * @return SchemaBuilder
+     */
+    public static function binary($length = null)
+    {
+        return static::createDefault(Schema::TYPE_BINARY, $length);
+    }
+
+    /**
+     * Makes column a boolean
+     *
+     * @param integer $length
+     * @return SchemaBuilder
+     */
+    public static function boolean($length = null)
+    {
+        return static::createDefault(Schema::TYPE_BOOLEAN, $length);
+    }
+
+    /**
+     * Makes column a money
+     *
+     * @param integer $precision
+     * @param integer $scale
+     * @return SchemaBuilder
+     */
+    public static function money($precision = null, $scale = null)
+    {
+        return static::createNumeric(Schema::TYPE_MONEY, $precision, $scale);
+    }
+
+   /**
+    * Specify value column could be null
+    *
+    * @return SchemaBuilder
+    */
+    public function notNull()
+    {
+        $this->isNotNull = true;
+
+        return $this;
+    }
+
+    /**
+     * Calls the named method which is not a class method.
+     *
+     * Do not call this method directly as it is a PHP magic method that
+     * will be implicitly called when an unknown method is being invoked.
+     * @param string $name the method name
+     * @param array $arguments method parameters
+     * @return mixed the method return value
+     */
+    public function __call($name, $arguments)
+    {
+        if ($name === 'default') {
+            return call_user_func_array([$this, '_default'], $arguments);
```

Comment:
We discussed about it here. We want write method with name 'default', but it reserved in PHP.